src/Pages/Components/Form.jsx

```diff
@@ -0,0 +1,130 @@
+
+function Copyright(props) {
+  return (
+    <Typography
+      variant="body2"
+      color="text.secondary"
+      align="center"
+      {...props}
+    >
+      {"Copyright © "}
+      <Link color="inherit" href="https://mui.com/">
+        Your Website
+      </Link>{" "}
+      {new Date().getFullYear()}
+      {"."}
+    </Typography>
+  );
+}
+
+const theme = createTheme();
+
+export default function Form({
+  handleOnChange,
+  handleSubmit,
+  action,
+  password,
+  userName,
+}) {
+  return (
+    <ThemeProvider theme={theme}>
+      <Container component="main" maxWidth="xs">
+        <CssBaseline />
+        <Box
+          sx={{
+            marginTop: 8,
+            display: "flex",
+            flexDirection: "column",
+            alignItems: "center",
+          }}
+        >
+          <Avatar sx={{ m: 1, bgcolor: "secondary.main" }}>
+            <LockOutlinedIcon />
+          </Avatar>
+          <Typography component="h1" variant="h5">
+            {action}
+          </Typography>
+          <Box
+            component="form"
+            noValidate
+            onSubmit={handleSubmit}
+            sx={{ mt: 3 }}
+          >
+            <Grid container spacing={2}>
+              <Grid item xs={12}>
+                <TextField
+                  fullWidth
+                  id="name"
+                  label="User name *"
+                  name="name"
+                  value={userName}
+                  onChange={handleOnChange}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  required
+                  fullWidth
+                  name="password"
+                  label="Password"
+                  type="password"
+                  id="password"
+                  value={password}
+                  onChange={handleOnChange}
+                />
+              </Grid>
+              <Grid item xs={12}>
+                <FormControlLabel
+                  control={
+                    <Checkbox value="allowExtraEmails" color="primary" />
+                  }
+                  label="I agree with whatever."
+                />
+              </Grid>
+            </Grid>
+            <Button
+              type="submit"
+              fullWidth
+              variant="contained"
+              sx={{ mt: 3, mb: 2 }}
+              // onClick={() =>
+              //   setAction(action === "Login" ? "Sign up 1" : "Sign up 2")
+              // }
+            >
+              {action}
+            </Button>
+            <Grid container justifyContent="flex-end">
+              <Grid item>
+                {action === "Login" ? (
+                  <Link href="/sign-up" variant="body2">
+                    Don't have an account? Sign Up
+                  </Link>
+                ) : (
+                  <Link href="/login" variant="body2">
+                    Already have an account? Login
+                  </Link>
```

Comment:
It would be a good idea to use "Link" from `react-router-dom` here as well.

The styling shouldn't be affected too much.

https://v5.reactrouter.com/web/api/Link